VEDA_OS01/tests/test_transcripts.py

```diff
@@ -1377,3 +1380,184 @@ def test_translations_retrieval_exceptions(self, mock_responses, expected_loggin
+            status(int): HTTP status code.
+            data(dict): Dict of data to mock
+
+        Returns:
+            mock_object(Mock) - Modified mocked object.
+        """
+        mock_object = Response(status=status)
+        mock_object.content = json.dumps(data)
+        return mock_object
+
+    def test_transcript_credentials_not_allowed(self):
+        """
+        Tests that GET method is not allowed.
+        """
+        response = self.client.get(self.url)
+        response_status_code = response.status_code
+        response = json.loads(response.content)
+
+        self.assertEqual(response_status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+        self.assertEqual(response['detail'], 'Method "GET" not allowed.')
+
+    def test_transcript_credentials_unauthorized(self):
+        """
+        Tests that if user is not logged in we get Unauthorized response.
+        """
+        # Logut client if previously logged in.
+        self.client.logout()
+
+        # Try to send post without being authorized / logged in.
+        response = self.client.post(self.url, data=json.dumps({'org': 'test'}), content_type='application/json')
+        response_status_code = response.status_code
+        response = json.loads(response.content)
+
+        self.assertEqual(response_status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertEqual(response['detail'], 'Authentication credentials were not provided.')
+
+    @data(
+        (
+            {},
+            'org and provider and api_key'
+        ),
+        (
+            {
+                'org': 'test'
+            },
+            'provider and api_key'
+        ),
+        (
+            {
+                'org': 'test',
+                'provider': TranscriptProvider.CIELO24,
+                'api_key': 'test-api-key'
+            },
+            'cielo_username'
+        ),
+        (
+            {
+                'org': 'test',
+                'provider': TranscriptProvider.THREE_PLAY,
+                'api_key': 'test-api-key'
+            },
+            'three_play_secret_key'
+        ),
+        (
+            {
+                'org': 'test',
+                'provider': TranscriptProvider.CIELO24,
+                'api_key': 'test-api-key',
+                'cielo_username': 'test-cielo-user',
+                'three_play_secret_key': ''
+            },
+            None
+        ),
+        (
+            {
+                'org': 'test',
+                'provider': TranscriptProvider.THREE_PLAY,
+                'api_key': 'test-api-key',
+                'cielo_username': '',
+                'three_play_secret_key': 'test-secret-key'
+            },
+            None
+        )
+    )
+    @unpack
+    @patch('VEDA_OS01.transcripts.LOGGER')
+    @patch('VEDA_OS01.transcripts.TranscriptCredentialsView.get_cielo_token_response')
+    def test_transcript_credentials(self, post_data, missing_keys, get_cielo_token_mock, mock_logger):
```

Comment:
we should break it to  `test_transcript_credentials_missing_params` and `test_transcript_credentials_happy_flow`.